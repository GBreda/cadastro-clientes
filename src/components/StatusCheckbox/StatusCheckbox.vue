<template>
  <div class="d-flex align-items-center">
    <input type="checkbox" id="checkbox" v-model="checked" />
    <label for="checkbox">Ativo</label>
  </div>
</template>

<script setup>
const checked = defineModel();
</script>
