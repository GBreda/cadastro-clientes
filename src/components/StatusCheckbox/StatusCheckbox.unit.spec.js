import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import StatusCheckbox from './StatusCheckbox.vue';

const mountStatusCheckbox = () => {
  return mount(StatusCheckbox);
};

describe('StatusCheckbox', () => {
  let wrapper;

  it('should mount the component', () => {
    wrapper = mountStatusCheckbox();

    expect(wrapper.vm).toBeDefined();
  });

  it('should match the snapshot', () => {
    wrapper = mountStatusCheckbox();

    expect(wrapper.element).toMatchSnapshot();
  });
});
