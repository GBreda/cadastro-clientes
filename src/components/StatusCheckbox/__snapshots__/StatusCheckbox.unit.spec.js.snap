// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StatusCheckbox > should match the snapshot 1`] = `
<div
  class="d-flex align-items-center"
>
  <input
    id="checkbox"
    type="checkbox"
  />
  <label
    for="checkbox"
  >
    Ativo
  </label>
</div>
`;
